drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,313 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+		if err != nil {
+			return false, err
+		}
+		if !replicaColocated {
+			return false, fmt.Errorf("replicaset mismatch for volume [%s]", vol.ID)
+		}
+		isBindMounted := false
+		t := func() (interface{}, bool, error) {
+			vmPod, err := getVMPod(virtualMachineCtx, vol)
+			if err != nil {
+				// this is expected while the live migration is running since there will be 2 VM pods
+				log.Infof("Could not get VM pod for %s for context %s: %v", vol.Name, virtualMachineCtx.App.Key, err)
+				return false, false, nil
+			}
+			log.Infof("Verifying bind mount for %s", vol)
+			diskName := ""
+			for _, vmVol := range vmPod.Spec.Volumes {
+				pvcName := volInspect.Locator.VolumeLabels["pvc"]
+				if vmVol.PersistentVolumeClaim != nil && vmVol.PersistentVolumeClaim.ClaimName == pvcName {
+					diskName = vmVol.Name
+					break
+				}
+			}
+			mountType, err := getVMDiskMountType(vmPod, vol, volumeDriver, diskName)
+			if err != nil {
+				log.Warnf("Failed to get mount type of %s for context %s: %v", vol, virtualMachineCtx.App.Key, err)
+				return false, false, nil
+			}
+			log.Infof("Mount type of %s for context %s: %s", vol.Name, virtualMachineCtx.App.Key, mountType)
+			if mountType != mountTypeBind {
+				if !initialCheck {
+					log.Warnf("Waiting for %s for context %s to switch to bind-mount from %q",
+						vol.Name, virtualMachineCtx.App.Key, mountType)
+				}
+				return false, false, nil
+			}
+			isBindMounted = true
+			return true, false, nil
+		}
+		if initialCheck {
+			// initial check is done only once
+			_, _, err := t()
+			if err != nil {
+				return false, err
+			}
+			if !isBindMounted {
+				return false, fmt.Errorf("volume [%s] is not bind mounted", vol.ID)
+			}
+			continue // skip retrying
+		} else {
+			_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+	log.Infof("Successfully verified bind mount for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	return true, nil
+}
+
+// GetNodeOfVM returns nodename on which VM is running
+func GetNodeOfVM(virtualMachineCtx kubevirtv1.VirtualMachine) (string, error) {
+	vmi, err := kubevirt.Instance().GetVirtualMachineInstance(context1.TODO(), virtualMachineCtx.Name, virtualMachineCtx.Namespace)
+	if err != nil {
+		return "", err
+	}
+	log.InfoD("NodeName: %s", vmi.Status.NodeName)
+	return vmi.Status.NodeName, nil
+}
+
+// isNodeInReplicaSet verifies if the replicaset nodes are present in global replicaset
+func IsNodeInReplicaSet(replicaset []*api.ReplicaSet, globalReplicSet []*api.ReplicaSet) (bool, error) {
+	for _, rs := range replicaset {
+		found := false
+	outerLoop:
+		for _, rsNode := range rs.Nodes {
+			for _, grs := range globalReplicSet {
+				if len(grs.Nodes) != len(rs.Nodes) {
+					return false, fmt.Errorf("replicaset mismatch node count mismatch")
```

Comment:
include more context in the error message

Reply:
Done. Added more info.